x/blob/keeper/keeper.go

```diff
@@ -58,7 +53,9 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 func (k Keeper) PayForBlob(goCtx context.Context, msg *types.MsgPayForBlob) (*types.MsgPayForBlobResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	gasToConsume := uint64(shares.MsgSharesUsed(int(msg.BlobSize)) * GasPerMsgShare)
+	// calculate gas per message share by fetching the constant share size and the gas cost per byte from the KV store
+	gasPerMsgShare := appconsts.ShareSize * k.GasPerBlobByte(ctx)
+	gasToConsume := uint64(shares.MsgSharesUsed(int(msg.BlobSize)) * int(gasPerMsgShare))
```

Comment:
[question] Is it possible to include a unit test that verifies the gas consumed now depends on the param? Perhaps the unit test can determine the gas cost of transaction `k.GasPerBlobByte(ctx) = 8` and `k.GasPerBlobByte(ctx) = 9` and verify the latter is > the former.

If a unit test is is difficult to construct, can this please be manually test to confirm it works as expected?